Runtime/Scripts/RGBotConfigs/RGState.cs

```diff
@@ -39,24 +41,28 @@ public RGStateEntity GetGameObjectState()
         {
             var theTransform = rgEntity.transform;
             
-            var state = new RGStateEntity()
-            {
-                ["id"] = theTransform.GetInstanceID(),
-                ["type"] = rgEntity.objectType,
-                ["isPlayer"] = rgEntity.isPlayer,
-                ["isRuntimeObject"] = rgEntity.isRuntimeObject,
-				["position"] = theTransform.position,
-				["rotation"] = theTransform.rotation,
-            };
-
+            var state = CreateStateEntity();
+            state["id"] = theTransform.GetInstanceID();
+            state["type"] = rgEntity.objectType;
+            state["isPlayer"] = rgEntity.isPlayer;
+            state["isRuntimeObject"] = rgEntity.isRuntimeObject;
+            state["position"] = theTransform.position;
+            state["rotation"] = theTransform.rotation;
+            
             var dict = GetState();
             foreach (var entry in dict)
             {
-                state.Add(entry.Key, entry.Value);
+                // allow overriding default state fields like position
+                state[entry.Key] = entry.Value;
             }
 
             return state;
         }
+
+        protected virtual RGStateEntity CreateStateEntity()
+        {
+            return new RGStateEntity();
```

Comment:
What exactly is this?

Reply:
Should we add this new method to the docs?

Reply:
I'll add some code level docs, but this becomes important for complex internal state objects to give you a fully typed StateEntity to work with in C# instead of having to dig through a dictionary.

Reply:
resolved